### PR TITLE
build(fluid-runner): Enable API docs and make CJS-only package

### PR DESCRIPTION
**Pull request by tylerbutler**
Updates the fluid-runner package to build API docs. There are several API inconsistencies that are ignored in config for now.

The package is now explicitly CJS-only (it wasn't building ESM anyway).

I won't merge this until after the 8.0 release branch is created.

### CODE REVIEW

File: packages/tools/fluid-runner/api-extractor.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+	"$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
+	"extends": "../../../common/build/build-common/api-extractor-base.json",
+	"messages": {
+		"extractorMessageReporting": {
+			// TODO: remove once base config has this enabled as an error
+			"ae-forgotten-export": {
+				"logLevel": "warning",
+				"addToApiReportFile": false
```
Comment (Josmithr) [inline packages/tools/fluid-runner/api-extractor.json:9]: I would probably make this true until we promote it to an error
```suggestion
				"addToApiReportFile": true
```

File: packages/tools/fluid-runner/api-extractor.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			},
+			"ae-undocumented": {
+				"logLevel": "warning",
+				"addToApiReportFile": false
+			},
+			"ae-unresolved-link": {
+				"logLevel": "warning",
+				"addToApiReportFile": false
```
Comment (Josmithr) [inline packages/tools/fluid-runner/api-extractor.json:17]: I would probably make this true until we promote it to an error
```suggestion
				"addToApiReportFile": true
```

**Review by Josmithr — APPROVED** [review]
Left a couple of suggestions, but otherwise looks good!

File: packages/tools/fluid-runner/package.json [hunk 3]
```diff
@@ -10,18 +10,29 @@
 	},
 	"license": "MIT",
 	"author": "Microsoft and contributors",
+	"type": "commonjs",
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "default": "./dist/index.js"
+    }
+  },
```
Comment (alexvy86) [inline packages/tools/fluid-runner/package.json:19]: Funky indentation

**Pull request merged**